ssl/ssl_lib.c

```diff
@@ -4360,6 +4361,72 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx)
     return ctx->ctlog_store;
 }
 
+void SSL_CTX_set_early_cb(SSL_CTX *c, SSL_early_cb_fn cb, void *arg)
+{
+    c->early_cb = cb;
+    c->early_cb_arg = arg;
+}
+
+int SSL_early_isv2(SSL *s)
+{
+    return s->clienthello->isv2;
+}
+
+unsigned int SSL_early_get0_legacy_version(SSL *s)
+{
+    return s->clienthello->legacy_version;
```

Comment:
:)
I put the check into SSL_early_get0_ex() largely to prompt this question, namely, is the lack of s->clienthello an internal error to be bubbled up to the application, or an application error, for calling an SSL_early_* API from an invalid context?  I am inclined to go with the latter, since the handshake ends up in an error state if we fail to set s->clienthello to a valid value (okay, I had to go add the NULL check just now, but it's clear it should have always been there) and it should stay there through any valid codepaths that end up in the early callback being called.
That is, I favor removing the check from SSL_early_get0_ext() and asserting that the SSL_early() functions cannot fail [in correct usage].